src/core/ext/filters/client_channel/connector.h

```diff
@@ -48,16 +51,22 @@ class SubchannelConnector : public InternallyRefCounted<SubchannelConnector> {
     ChannelArgs channel_args;
   };
 
+  struct TransportDeleter {
+    void operator()(grpc_transport* transport) {
+      grpc_transport_destroy(transport);
+    }
+  };
+  using TransportPointer = std::unique_ptr<grpc_transport, TransportDeleter>;
   struct Result {
     // The connected transport.
-    grpc_transport* transport = nullptr;
+    TransportPointer transport;
     // Channel args to be passed to filters.
     ChannelArgs channel_args;
     // Channelz socket node of the connected transport, if any.
     RefCountedPtr<channelz::SocketNode> socket_node;
 
     void Reset() {
-      transport = nullptr;
+      transport.reset();
```

Comment:
Instead of using `unique_ptr<>` here, I suggest just changing this to the following:

```
if (transport != nullptr) {
  grpc_transport_destroy(transport);
  transport = nullptr;
}
```

I think it will make more sense to use a smart pointer in the future, when we change the transport API to be an idiomatic C++ API.

Reply:
Makes sense! Done. Thanks for the review!